### PR TITLE
Issue #5930: extend Torch 2.13.0 CPU safety guard to Python 3.11 (cheap-lane worker)

### LINKED ISSUE #5930: friction: PPO-Mamba optional readiness still exits with SIGSEGV
## Goal / Problem

**Observed behavior**

The full local readiness command exits in the optional lane because the PPO-Mamba dry-run test terminates with signal 11:

```text
tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_dry_run_emits_parameter_summary_placeholder
CompletedProcess.returncode == -11
```

This recurs after the prior friction tracker #5724 was closed following #5739. The failure is outside PR #5923’s changed files, but it prevents the configured repository readiness gate from completing.

**Expected behavior**

The focused PPO-Mamba dry-run should complete successfully or report an actionable fail-closed availability result without a native crash.

## Archetype Metadata

```yaml
archetype: workflow
evidence_tier: smoke
linked_policy:
  - docs/maintainer_values.md
  - docs/dev_guide.md
```

## Reproduction

```bash
BASE_REF=origin/main scripts/dev/pr_ready_check.sh
uv run pytest tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_dry_run_emits_parameter_summary_placeholder -q
```

Observed on 2026-07-17 in the PR-gate worktree with Python 3.11 and the repository `.venv`; the full optional lane had 3,648 passed and 2 skipped before this failure.

## Scope

- Affected module(s): `scripts/training/train_ppo.py`, `robot_sf/feature_extractors/mamba_extractor.py`, optional readiness/test environment.
- Regression: recurrence of #5724’s failure mode after #5739.
- In scope: identify the native extension/runtime combination and make this smoke path complete or fail cleanly.
- Out of scope: changes to benchmark analyzer PR #5923 and unrelated training redesign.

## Definition of Done

- [ ] The focused PPO-Mamba dry-run completes successfully or emits an actionable fail-closed availability result rather than signal 11.
- [ ] The optional readiness lane records the outcome without a native worker crash.
- [ ] A regression check covers the discovered runtime combination or guard.
- [ ] The result and any compatibility boundary are documented.

## Validation / Testing

- [ ] Run the focused reproduction command above serially.
- [ ] Run `BASE_REF=origin/main scripts/dev/pr_ready_check.sh` through the optional lane.
- [ ] Confirm the changed path does not weaken the checker or silently convert a real smoke failure into success.

## Risk Assessment

- Functional risk: optional training smoke can be misclassified or crash the readiness lane.
- Compatibility risk: native torch/SSM/TensorFlow-adjacent dependencies may interact differently across hosts.
- Mitigation: preserve explicit diagnostic-only status and fail closed when the required runtime is unavailable.

## Related

- Supersedes/recurs: #5724
- Observed by gate for: #5923

### PR DESCRIPTION
**Pull request by ll7**
## What/Why

When running tests or scripts that load `stable_baselines3` under Python 3.11 with `torch==2.13.0`, the subsequent call to `torch.use_deterministic_algorithms(True)` imports `torch._inductor.config` (reaching Dynamo/Triton) and causes a segmentation fault (`signal 11`, exit code 139) on CPU-only/stub CUDA configurations. This happens because TensorFlow/Tensorboard and Triton load conflicting CUDA/stub library symbols when TensorFlow/Tensorboard is initialized first.

This was previously fixed for Python 3.12+ in #5739 (for #5724) by skipping Triton preloading and setting `TORCH_COMPILE_DISABLE=1`. However, Python 3.11 is also vulnerable under the same conditions. We extended the runtime safety guard in `robot_sf/common/seed.py` to target Python 3.11+ as well (i.e. changing `sys.version_info[:2] >= (3, 12)` to `sys.version_info[:2] >= (3, 11)`).

### Pytest Execution (Python 3.11)

```text
$ uv run --python 3.11 pytest tests/training/test_train_ppo_mamba_config_issue_4014.py -v
============================= test session starts ==============================
platform linux -- Python 3.11.15, pytest-9.1.1, pluggy-1.6.0
...
tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_smoke_config_loads_cpu_safe_extractor_contract PASSED [ 33%]
tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_policy_selection_routes_mamba_extractor PASSED [ 66%]
tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_dry_run_emits_parameter_summary_placeholder PASSED [100%]
============================== 3 passed in 8.30s ===============================
```

### Pytest Seed Utils (Python 3.11)

```text
$ uv run --python 3.11 pytest tests/test_seed_utils.py -v
...
============================== 14 passed in 5.75s ==============================
```

### Ruff Validation

```text
$ uv run ruff check robot_sf/common/seed.py tests/test_seed_utils.py
All checks passed!

$ uv run ruff format --check robot_sf/common/seed.py tests/test_seed_utils.py
2 files already formatted
```

## Successor Statement

This PR is a successor slice that does not duplicate prior work and fully resolves issue #5930 by extending the safety check for conflicting modules to Python 3.11.

This PR was produced by the agy/Gemini-3.5-Flash cheap implementation lane; the review gate hardens and merges.

Closes #5930

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved compatibility for Torch 2.13.0 on Python 3.11 and newer.
  * Prevented deterministic execution and runtime initialization issues on supported Python versions.

* **Tests**
  * Expanded compatibility coverage for Python 3.11, 3.12, and 3.13.
  * Verified that older Python versions remain unaffected.

### CODE REVIEW

**Comment by ll7** [comment]
Gate findings for PR #5939 at observed head `5195562facce61fabc92be66ee2f724abf488054`.

Review basis: current PR diff, linked issue #5930 and its current body, current review body, current checks, and current review-thread snapshot.

Open review threads: 0 open review threads (GraphQL review-thread snapshot).

1. CodeRabbit review finding in the open review body (not represented as an inline GraphQL thread): update the surrounding Torch 2.13 compatibility docstrings in `robot_sf/common/seed.py` to say Python 3.11+ and rename the affected test in `tests/test_seed_utils.py` so its name reflects Python 3.11–3.13. Verdict: fixed — phase 2 will apply this documentation/test-name-only repair.

The Gemini quota notice is advisory and contains no code finding. No additional correctness, scope, placeholder-docstring, or closure findings from the gate review. The PR remains non-merge-ready until the finding is implemented, tested, pushed, and rechecked.

gate-verdict: changes-requested @ 5195562facce61fabc92be66ee2f724abf488054

**Comment by ll7** [comment]
Gate final findings for PR #5939 at current head `83324d18692b53326b568e37687029710261271a`.

Delta review: the refresh merge `83324d18692b53326b568e37687029710261271a` merged current `origin/main`; the prior repair at `2c1fd33757e7321631cea0ecb28a897c452bac5b` remains present and no new finding was introduced.

Open review threads: 0 open review threads (GraphQL review-thread snapshot).

1. CodeRabbit’s open top-level review finding (not represented as an inline GraphQL thread): compatibility docstrings still said Python 3.12+, and the determinism test name still omitted Python 3.11. Verdict: fixed in `2c1fd33757e7321631cea0ecb28a897c452bac5b`; the two docstrings now say Python 3.11+ and the test name reflects Python 3.11+.

The Gemini quota notice was advisory and contained no code finding. No additional correctness, scope, placeholder-docstring, closure, or provenance findings.

Contract/closure: issue #5930 is fully addressed; `Closes #5930` is accurate. The guard, regression coverage, compatibility wording, and optional smoke path are covered. No parent issue requires propagation.

Validation:
- `uv run pytest tests/test_seed_utils.py -q` — 14 passed at the refreshed head.
- `uv run pytest tests/training/test_train_ppo_mamba_config_issue_4014.py -q` with the declared `training` extra — 3 passed.
- Targeted Ruff check, format check, and `git diff --check` — passed.
- Current-head CI run `29564892841` — 15/15 jobs successful; `gh pr checks` — all pass.
- Merge staleness — FRESH against `origin/main` `c4899978d2a6e6c442fed6ca74a5d095e7ee666e`; merge hold — allowed.
- Local `output/compute_feasibility_matrix_v1/` and `output/repos/` are ignored worktree-local cache/evidence outputs; no durable artifact is required by this PR.

gate-verdict: accepted @ 83324d18692b53326b568e37687029710261271a
